src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2050,11 +2050,6 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
       clone
     }
 
-    private[this] object updateAttachmentFun extends Function1[Any, Unit] {
-      var theClone: TypeOfClonedSymbol = null
-      def apply(x: Any): Unit = theClone.updateAttachment(x)
-    }
-
```

Comment:
For future reference, I realised that this was defined at the wrong level -- it should have been a member of `Symbols` not of `Symbol`.